### PR TITLE
Signature help should use SymbolDisplayMiscellaneousOptions.AllowDefaultLiteral

### LINKED ISSUE #47364: Signature help should use SymbolDisplayMiscellaneousOptions.AllowDefaultLiteral
image

It's less noisy if IntelliSense just shows `default` instead of `default(CancellationToken)`.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #47364

### CODE REVIEW

File: src/Features/CSharp/Portable/SignatureHelp/AbstractOrdinaryMethodSignatureHelpProvider.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ internal abstract class AbstractOrdinaryMethodSignatureHelpProvider : AbstractCS
                 result.Add(Space());
             }
 
-            result.AddRange(method.ToMinimalDisplayParts(semanticModel, position, MinimallyQualifiedWithoutParametersFormat));
+            result.AddRange(method.ToMinimalDisplayParts(semanticModel, position, MinimallyQualifiedWithoutParametersFormat.AddMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.AllowDefaultLiteral)));
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/SignatureHelp/AbstractOrdinaryMethodSignatureHelpProvider.cs:82]: this is the 'without parameters' portion.  i.e. it affects the part of the signature up *to* the `(`.  so it wouldn't/shouldn't affect parameters.

Comment (Youssef1313) [reply src/Features/CSharp/Portable/SignatureHelp/AbstractOrdinaryMethodSignatureHelpProvider.cs:82]: @CyrusNajmabadi Oh. Thanks for pointing this out.

Probably I should start debugging in https://github.com/dotnet/roslyn/blob/4d24556df984774dc70cf707eff1ee9d3d96b9b9/src/Features/CSharp/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.cs

Comment (Youssef1313) [reply src/Features/CSharp/Portable/SignatureHelp/AbstractOrdinaryMethodSignatureHelpProvider.cs:82]: @CyrusNajmabadi I added a test in InvocationExpressionSignatureHelpProviderTests:

```
        [Fact, Trait(Traits.Feature, Traits.Features.SignatureHelp)]
        public async Task TestInvocationOnTriggerParens_OptionalDefaultStruct()
        {
            var markup = @"
class Program
{
   static void SomeMethod(CancellationToken token = default) => throw new NotImplementedException();

    static void Main(string[] args)
    {
        [|SomeMethod($$|]);
    }
}";

            var expectedOrderedItems = new List<SignatureHelpTestItem>();
            expectedOrderedItems.Add(new SignatureHelpTestItem("void Program.SomeMethod([CancellationToken token = default])", string.Empty, null, currentParameterIndex: 0));

            await TestAsync(markup, expectedOrderedItems, usePreviousCharAsTrigger: true);
        }
```

It fails with the following:

```
  Error Message:
   Assert.Equal() Failure
                                  (pos 51)
Expected: ···lationToken token = default])
Actual:   ···lationToken token = null])
                                  (pos 51)
```

The actual result from the test doesn't match the actual result that actually happens, and also doesn't match the expected behavior. Am I testing the wrong thing? 😄 

In any case, it's weird to see `CancellationToken token = null` as it's a non-nullable value type.

Comment (Youssef1313) [reply src/Features/CSharp/Portable/SignatureHelp/AbstractOrdinaryMethodSignatureHelpProvider.cs:82]: That's a funny mistake :smile:
I'll fix that and continue trying to debug this locally.
Thanks a lot for pointing to my mistake.

**Comment by Youssef1313** [comment]
The test passes locally now. This is ready for review.

File: src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ protected static SymbolDisplayPart NewLine()
                 parameter.Name,
                 parameter.IsOptional,
                 parameter.GetDocumentationPartsFactory(semanticModel, position, formatter),
-                parameter.ToMinimalDisplayParts(semanticModel, position));
+                parameter.ToMinimalDisplayParts(semanticModel, position, SymbolDisplayFormat.MinimallyQualifiedFormat.AddMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.AllowDefaultLiteral)));
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs:56]: can you make this a static variable so we don't have to allocate a fresh instance on each param?  thanks :)

File: src/EditorFeatures/CSharpTest/SignatureHelp/InvocationExpressionSignatureHelpProviderTests.cs [hunk 3]
```diff
@@ -962,6 +962,29 @@ void Goo(int someParameter, bool something)
 
         #region "Trigger tests"
 
+        [Fact, Trait(Traits.Feature, Traits.Features.SignatureHelp)]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/SignatureHelp/InvocationExpressionSignatureHelpProviderTests.cs:965]: WorkItem

**Review by CyrusNajmabadi — APPROVED** [review]
LGTM with hte PR feedback taken care of.  please ping me when green :)

File: src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs [hunk 4]
```diff
@@ -13,6 +13,9 @@ namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp
 {
     internal abstract class AbstractCSharpSignatureHelpProvider : AbstractSignatureHelpProvider
     {
+        private static SymbolDisplayFormat s_allowDefaultLiteralFormat = SymbolDisplayFormat.MinimallyQualifiedFormat
```
Comment (Youssef1313) [inline src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs:16]:
```suggestion
        private static readonly SymbolDisplayFormat s_allowDefaultLiteralFormat = SymbolDisplayFormat.MinimallyQualifiedFormat
```

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs [hunk 5]
```diff
@@ -53,7 +56,7 @@ protected static SymbolDisplayPart NewLine()
                 parameter.Name,
                 parameter.IsOptional,
                 parameter.GetDocumentationPartsFactory(semanticModel, position, formatter),
-                parameter.ToMinimalDisplayParts(semanticModel, position));
+                parameter.ToMinimalDisplayParts(semanticModel, position, s_allowDefaultLiteralFormat));
```
Comment (davidwengier) [inline src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs:59]: Does this need to take into account language version, so we don't show default literals to people on C# <7.1? Or do we not worry about that sort of thing? Or does ToMinimalDisplayParts take care of that?

Comment (Youssef1313) [reply src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs:59]: @davidwengier I don't know if ToMinimalDisplayParts accounts for language version or not.
Probably @sharwell or @CyrusNajmabadi know?

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs:59]: It doesn't need to account for lang version.  How we present info is different from what suggestions we offer, or what code we generate :)

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**